peps/pep-0775.rst

```diff
@@ -0,0 +1,136 @@
+PEP: 775
+Title: Make zlib required to build Python
+Author: Stan Ulbrych <stanulbrych@gmail.com>,
+        Petr Viktorin <encukou@gmail.com>
+Discussions-To: https://discuss.python.org/t/lets-make-zlib-required-rather-than-optional-to-build-cpython/23062
+Status: Draft
+Type: Standards Track
+Created: 20-Feb-2025
+Python-Version: 3.14
+Post-History: `23-Jan-2023 <https://discuss.python.org/t/23062>`__
+
+Abstract
+========
+
+Building CPython on systems without the `zlib <https://zlib.net>`_ compression library, except WASI,
+is no longer supported.
+
+The ``zlib`` module is made a required part of the standard library,
+except on WASI.
+
+Builds without ``zlib`` will still be possible, but formally unsupported.
+
+
+Motivation
+==========
+
+The ``zlib`` library, which powers the ``zlib`` Python module,
+is available on virtually all systems.
+
+Many wheels on PyPI, including the `pip`_ installer, require ``zlib``.
+Users of *pip* would consider CPython without ``zlib`` to be broken,
+but mostly don't notice because all major builds of CPython include ``zlib``.
+
+.. _pip: https://pypi.org/project/pip/
+
+CPython developers don't really notice either. It turns out that at the time
+of this writing, at least one CPython test fails without ``zlib`` (the “skip”
+decorator in ``test_peg_generator.test_c_parser`` is applied too late),
+but our CI didn't catch this.
+
+This PEP treats this as an issue in documentation and messaging.
+In practice, we already don't support building CPython without ``zlib``; we
+should just say so.
+
+
+Rationale
+=========
+
+There are possible use cases for zlib-less builds, such as embedding and
+bootstrapping, as well as unforeseen ones.
+Therefore, we don't *remove* support for zlib-less systems; we mark them
+unsupported and invite affected users to do their own testing, or to share
+use cases that can make us reconsider this decision.
+
+``zlib`` is not yet used by default on the WASI platform -- mostly because
+adding it hasn't yet been a priority there. (Note that `Pyodide`_, the main
+“real-world” CPython distribution for WASI, does include ``zlib``.)
+We take this as an opportunity to  continue testing a platform without
+``zlib``, so that we don't unintentionally break unsupported builds yet.
```

Comment:
```suggestion
zlib is not yet used by default on the WASI platform -- mostly because
adding it hasn't yet been a priority there. (Note that `Pyodide`_, the main
"real-world" CPython distribution for WASI, does include zlib.)
We take this as an opportunity to  continue testing a platform without
zlib, so that we don't unintentionally break unsupported builds yet.
```